src/include/daos/common.h

```diff
@@ -517,4 +517,16 @@ daos_unparse_ctype(daos_cont_layout_t ctype, char *string)
 	}
 }
 
+static inline int get_current_time(uint64_t *time)
+{
+	struct timespec	now;
+	int		rc;
+
+	rc = clock_gettime(CLOCK_MONOTONIC_COARSE, &now);
+	if (rc == 0)
+		*time = d_time2s(now);
```

Comment:
If you want an accurate time, you'd call d_timeus_secdiff(0) directly; If you want a coarse time, the 'd_time2s' shouldn't be called to do the expensive calculation on 'tv_nsec'. (use the tv_sec only instead).

BTW, the function name could be changed to 'daos_gettime_corase()' if you want to make it as a general daos API?

Reply:
If you think d_time2s is too expensive, we can calculate it with some approximate value. For example:

(double) t.tv_sec + (double) t.tv_nsec >> 30;